.github/workflows/stag.yml

```diff
@@ -0,0 +1,22 @@
+name: deploy-to-staging
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - JainM6-patch3
```

Comment:
I would use some generic name for that, aso this branch is `JainM6-patch-3`

Reply:
Do you mean I should use a generic name for the branch `JainM6-patch3`? The purpose of this is to create a stag env. only for this branch.

Reply:
Sorry, there are two comments from me. Maybe I will separate it:

1. this branch is `JainM6-patch-3`, not `JainM6-patch3`, not sure if it's intended
2. You could create some general rule for branches, eg. all branches beginning with `stag/*` should be deployed on staging environment. However I don't know if there is need for that

Reply:
Yes, you are correct. I need to fix the branch name. For the 2. point, I don't think there is a need for that because I just wanted to deploy a specific PR to staging, not all. This workflow is just for one PR.